mantaray/views.py

```diff
@@ -542,6 +547,21 @@ def on_part(self, nick: str, reason: str | None) -> None:
             show_in_gui=self.server_view.should_show_join_leave_message(nick),
         )
 
+    def on_kick(self, channel: str, kicker_nick: str, kicked_nick: str, reason: str | None) -> None:
+        self.userlist.remove_user(kicked_nick)
+        if reason is None:
+            extra = ""
+        else:
+            extra = " " + reason + ""
+        self.add_message(
+            "*",
+            (kicker_nick, ["self-nick"]),
+            (' kicked ', []),
+            (kicked_nick, ["other-nick"]),
```

Comment:
There are 4 possible cases:
- Non-mantaray user kicks non-mantaray user (both are `other-nick`)
- Non-mantaray user kicks mantaray user (kicker `other-nick`, kicked `self-nick`)
- Mantaray user kicks non-mantaray user (kicker `self-nick`, kicked `other-nick`, like in this code)
- Mantaray user kicks himself (both `self-nick`)

You only handle one of the 4 possible cases.

Reply:
Should be working now